src/Bicep.Core/Semantics/ParamsSemanticModel.cs

```diff
@@ -39,10 +43,79 @@ public ParamsSemanticModel(BicepParamFile bicepParamFile, ImmutableArray<IDiagno
             // name binding is done
             // allow type queries now
             paramsSymbolContext.Unlock();
+            this.AssignmentsByDeclaration = InitializeDeclarationToAssignmentDictionary();
+            this.DeclarationsByAssignment = InitializeAssignmentToDeclarationDictionary();
             // lazy load single use diagnostic set
             this.AllDiagnostics = new Lazy<ImmutableArray<IDiagnostic>>(() => GetAllDiagnostics());
         }
 
+        public ImmutableArray<IDiagnostic> GetAllDiagnostics()
+            => BicepParamFile.ProgramSyntax.GetParseDiagnostics()
+                .Concat(ParamsTypeManager.GetAllDiagnostics())
+                .Concat(this.CompilationLoadDiagnostics)
+                .Concat(GetAdditionalSemanticDiagnostics()).ToImmutableArray();
+            
+        /// <summary>
+        /// Gets the file that was compiled.
+        /// </summary>
+        public ParamFileSymbol Root => this.ParamBinder.ParamFileSymbol;
+
+        public ParameterSymbol? GetParameterDeclaration(ParameterAssignmentSymbol parameterAssignmentSymbol)
+        {
+            return this.DeclarationsByAssignment[parameterAssignmentSymbol];
+        }
+
+        public ParameterAssignmentSymbol? GetParameterAssignment(ParameterSymbol parameterSymbol)
```

Comment:
Since the method returns a nullable type, it would be better to follow the convention and rename it to `TryGetParameterAssignment`